### PR TITLE
Add NMF for topic extraction

### PR DESCRIPTION
**Pull request by jilljenn**
Here is the non-negative matrix factorization, directly from scikit-learn, that allows more interpretable components. By definition, it approximates a non-negative matrix, so I decided to put a rating of 1 for like/favorite and 0 for the rest. Thus, it does not consider dislikes, it just tries to put together things people like.

The corresponding error values:
[ALS] 0.436 (still the best)
[NMF] 0.530
[ZERO] 0.649
[SVD] 0.761 (worse than the dumb algorithm saying 0 all the time)
(I should remove willsee and wontsee from the test set though :D)

BTW, there is a [sparseness parameter](http://scikit-learn.org/stable/modules/generated/sklearn.decomposition.NMF.html) in the scikit-learn implementation!

### CODE REVIEW

File: mangaki/mangaki/management/commands/compare.py [hunk 1]
```diff
@@ -24,7 +27,7 @@ class Experiment(object):
     results = {}
     algos = None
     def __init__(self, PIG_ID=None):
-        self.algos = [MangakiALS(20), MangakiSVD(20), MangakiKNN()]
+        self.algos = [MangakiNMF(30), MangakiALS(20), MangakiSVD(20), MangakiZero()]
```
Comment (Elarnon) [inline mangaki/mangaki/management/commands/compare.py]: No more KNN ?

File: mangaki/mangaki/utils/nmf.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+
+class MangakiNMF(object):
+    M = None
+    W = None
+    H = None
+    def __init__(self, NB_COMPONENTS=10):
+        self.NB_COMPONENTS = NB_COMPONENTS
+        self.chrono = Chrono(True)
+        self.works = pandas.read_csv('data/works.csv', header=None).as_matrix()[:, 1]
```
Comment (Elarnon) [inline mangaki/mangaki/utils/nmf.py]: Can we use `settings.BASE_DIR` to find the CSV here ? It'd be great to not have an implicit requirement on the directory you must be in to run `manage.py` commands.

Comment (RaitoBezarius) [inline mangaki/mangaki/utils/nmf.py]: By the way, we should move the `read_csv` and give them as a parameter, as all algorithms seems to use these files.

File: mangaki/mangaki/management/commands/compare.py (hunk 1 above)
Comment (jilljenn) [inline mangaki/mangaki/management/commands/compare.py]: Je le remets.

File: mangaki/mangaki/utils/values.py [hunk 3]
```diff
@@ -1 +1,2 @@
 rating_values = {'favorite': 4, 'like': 2, 'dislike': -2, 'neutral': 0.1, 'willsee': 0.5, 'wontsee': -0.5}
```
Comment (RaitoBezarius) [inline mangaki/mangaki/utils/values.py]: Tu es sûr que d'autres modules que la NMF ne sont pas dépendants de ça ?
